packages/vscode-extension/src/utilities/getMinimumSupportedNodeVersion.ts

```diff
@@ -18,15 +23,33 @@ export function getMinimumSupportedNodeVersion(appRoot: string): string {
   ) {
     const reactNativePackageJson = require(`${appRoot}/node_modules/react-native/package.json`);
     if (reactNativePackageJson.engines && reactNativePackageJson.engines.node) {
-      return reactNativePackageJson.engines.node;
+      reactNativeMinimumNodeVersion = reactNativePackageJson.engines.node;
     }
   }
 
-  // this is minimum supported version at the time this code was written, and therefore a default for this function,
-  // but this code should never be reached as react-native contains this information in its package json and reaching,
-  // this code suggests that the function was incorrectly called before installing node_modules.
-  Logger.warn(
-    "[Minimum Supported node version] Utility function called before node modules were installed, the returned value might, be deprecated."
-  );
-  return ">=18";
+  const ranges = [
+    applicationMinimumNodeVersion,
+    reactNativeMinimumNodeVersion,
+    DEFAULT_MINIMUM_NODE_VERSION,
+  ].filter(Boolean);
+
+  if (ranges.length === 1) {
```

Comment:
This took me second to parse.
I think even something like
```suggestion
  if (!applicationMinimumNodeVersion && !reactNativeMinimumNodeVersion) {
```
communicates the intent better.